code-challenges.js

```diff
@@ -8,62 +8,98 @@
 
-const temperature2 = 350
-// Expected output: "350 is above boiling point"
 
-const temperature3 = 212
-// Expected output: "212 is at boiling point"
 
 // --------------------2) Create the code that will combine the two arrays and return the length using the test variables provided below.
 
-// Pseudo code:
+// // Pseudo code:Add a new variable that is able to be reassigned, give it a new meaningful name, link to the first const variable with an = assignment operator, concatinate it to the 2nd const variable. Console log the new variable name with the .length built in method to get the desired answer returned in the console log. 
 
-const padres1984WorldSeriesRuns = [2, 5, 2, 2, 4]
-const padres1998WorldSeriesRuns = [6, 3, 5, 3]
-// Expected output: 9
+// const padres1984WorldSeriesRuns = [2, 5, 2, 2, 4]
+// const padres1998WorldSeriesRuns = [6, 3, 5, 3]
+// // // Expected output: 9
+// let padresJoined = (padres1984WorldSeriesRuns).concat(padres1998WorldSeriesRuns)
+// console.log(padresJoined.length)
 
-// --------------------3) Create the code that will reverse the letters of a string using the test variable provided below.
 
-// Pseudo code:
 
-const currentCohort = "Echo 2023"
-// Expected output: "3202 ohcE"
 
-// --------------------4) Create the code that will return the last index of the given value from the array using the test variables provided below.
+// // --------------------3) Create the code that will reverse the letters of a string using the test variable provided below.
 
-// Pseudo code:
+// // Pseudo code: (I struggled to do this from lecture notes and had to go online for assistance) Add a brand new const variable with a name and make it an arrow function, now it's a function it is able to use different built in methods such as ".reverse". Console log the name of the new function, add the variable name holding the original string in brackets afterwards. It will return the string reversed! 
 
-const numberOfConnections = [13, 34, 42, 5, 5, 10, 27, 42, 10]
+// const currentCohort = "Echo 2023"
+// // Expected output: "3202 ohcE"
 
-const givenValue1 = 42
-// Expected output: 7
+// const reverseString = str => [...str].reverse().join('');
 
-const givenValue2 = 10
-// Expected output: 8
+// console.log(reverseString(currentCohort));
 
-// --------------------5) Create the code that will sort all the numbers in an array from largest to smallest. Use the test variables provided below.
+// // -------------------4) Create the code that will return the last index of the given value from the array using the test variables provided below.
 
-// Pseudo code:
+// // #1 - Pseudo code:change the const1 last of index to 6, console log const2 / by the changed last of index from const1.
+//#2 - Console log the var and  minus it by 2 to get output 8
+
+// const numberOfConnections = [13, 34, 42, 5, 5, 10, 27, 42, 10]
+
+// const givenValue1 = 42
+
+// numberOfConnections[8] = 6
+// console.log( givenValue1 /numberOfConnections[8])
```

Comment:
I think there's still some confusion on what the question is asking here. The prompt states the code should return the last index of the given value from the array. So looking at the given variables, we have `numberOfConnections` which is an array, and we have `givenValue1` and `givenValue2`, which are numbers. Looking at `givenValue1`, it's 42, so the question is asking at what index is the last appearance of 42 in the `numberOfConnections` array. If we look at the array, we can see the number 42 at index 2 and index 7, so the correct output should be 7, which was given in the expected output that was commented out. To do this, we can use the built-in method `.lastIndexOf()`, which will look through the array and return the index position of the last occurrence of whatever was passed into it as an argument.
`numberOfConnections.lastIndexOf(givenValue1)` will return 7.